### ISSUE TITLE
task id is too fuzzy

### ISSUE THREAD

**Issue by magicrobotmonkey** [dcos/dcos-cli#660]
Please answer the following questions before submitting your issue. Thanks!

### What version of DC/OS + DC/OS CLI are you using (`dcos --version`)?
`0.4.5`

### What operating system and version are you using?
```
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=14.04
DISTRIB_CODENAME=trusty
DISTRIB_DESCRIPTION="Ubuntu 14.04.3 LTS"
```

### What did you do?
If possible please provide a recipe for reproducing.
Launch a task with a very short id, then several others with that substr as part of their ids. 
Now you will not be able to view just that task. Same applies to viewing logs.

### What did you expect to see?
```
dcos task 33
NAME                HOST                 USER  STATE  ID
crashdeletes_job31  phx-r2-r4-comp21-hi  root    R    33
```

### What did you see instead?
```
dcos task 33
NAME                HOST                 USER  STATE  ID
ceph-v2.prometheus  phx-r2-r4-comp11-hi  root    R    prometheus_ceph-v2.14032612-33d5-11e6-b24e-0e46e10d898c
crashdeletes_job31  phx-r2-r4-comp21-hi  root    R    33
jupyter-smrtofastq  phx-r2-r4-comp22-hi  root    R    jupyter-smrtofastq.59a9d74c-334d-11e6-b24e-0e46e10d898c
mesos.prometheus    phx-r2-r4-comp8-hi   root    R    prometheus_mesos.063c8991-33d5-11e6-b24e-0e46e10d898c
vault-gatekeeper    phx-r2-r4-comp19-hi  root    R    vault-gatekeeper.21315fea-3321-11e6-b24e-0e46e10d898c
```

**Comment by jsancio** [comment]
@magicrobotmonkey thanks for the issue report. I know that we have a similar issue that we addressed for the next release. Can you confirm @tamarrow?

**Comment by tamarrow** [comment]
Yes, we have a fix in the next release, see pr #656
